NIPTool/API/external/api/api_v1/endpoints/batches.py

```diff
@@ -127,20 +148,33 @@ def fetal_fraction_XY(
     user: User = Depends(get_current_user),
 ):
     """Batch view with fetal fraction (X against Y) plot"""
+
     batch: Batch = find.batch(batch_id=batch_id, adapter=adapter)
 
-    control = get_ff_control_normal(adapter)
-    abnormal = get_ff_control_abnormal(adapter)
+    control: FetalFractionSamples = get_fetal_fraction.samples(
+        batch_id=batch_id, adapter=adapter, control_samples=True
+    )
+    abnormal: FetalFractionControlAbNormal = get_fetal_fraction.control_abnormal(adapter)
+    abnormal_dict = abnormal.dict(
+        exclude_none=True,
+        exclude={
+            "X0": {"status_data_"},
+            "XXX": {"status_data_"},
+            "XXY": {"status_data_"},
+            "XYY": {"status_data_"},
+        },
+    )
+
     return templates.TemplateResponse(
         "batch/tabs/FF_XY.html",
         context=dict(
             request=request,
             current_user=user,
             control=control,
-            abnormal=abnormal,
-            cases=get_ff_cases(adapter, batch_id),
-            max_x=max(control["FFX"]) + 1,
-            min_x=min(control["FFX"]) - 1,
+            abnormal=abnormal_dict,
```

Comment:
Yes. the abnormal is initially generated as a pydantic model: FetalFractionControlAbNormal. (line 157.) it is then transformed to a dict before bing passed to the template because it was more convinient in that way. In future i might rewrite the template to cope with the model instead.

Reply:
ok